searcher.py

```diff
@@ -13,66 +13,45 @@
 import nltk
 
 
-class CustomFilter(Filter):
-    # This filter will run for both the index and the query
-    is_morph = True
-    def __init__(self, filterFunc, *args, **kwargs):
-        self.customFilter = filterFunc
-        self.args = args
-        self.kwargs = kwargs
-    def __eq__(self):
-        return (other
-                and self.__class__ is other.__class__)
-    def __call__(self, tokens):
-        for t in tokens:
-            if t.mode == 'query': # if called by query parser
-                t.text = self.customFilter(t.text, *self.args, **self.kwargs)
-                yield t
-            else: # == 'index' if called by indexer
-                t.text = self.customFilter(t.text, *self.args, **self.kwargs)
-                yield t
-
-
 class Whoosher:
 
     def __init__(self, path="index/"):
         self.path=path
 
-    def set_schema(self):
+    def set_schema(self, df_schema):
+        """ Whoosh schema = all df_schema fields, stored but not indexed, 
+            + extra field 'body_processed', processed, indexed."""
         customWordFilter = RegexTokenizer() | \
                            LowercaseFilter() | \
                            CustomFilter(nltk.stem.porter.PorterStemmer().stem) | \
                            CustomFilter(nltk.WordNetLemmatizer().lemmatize)
 
-        return Schema(comment_ID=ID(stored=True),
-                      comment_Subreddit=ID(stored=True),
-                      comment_Content=TEXT(analyzer=customWordFilter),
-                      comment_Content_raw=STORED,
-                      )
+        whoosh_schema = {item:STORED for item in df_schema}

```

Comment:
field names not hardcoded anymore (except for the indexed one) and now coming from `df_schema` (to be set to `df.columns`).